'NoneType' object is not callable in v0.5.0-beta1.pex

## Summary

PEX start is interrupted.

## System information

 - Version: v0.5.0-beta1.pex 
 - Operating system: Ubuntu 16.10
 - Browser: n/a

## Traceback or relevant snippet from server.log or browser console

```
osboxes@osboxes:~$ cd Desktop/
osboxes@osboxes:~/Desktop$ ./kolibri-v0.5.0-beta1.pex start
INFO     Kolibri running for the first time.
INFO     We don't yet use pre-migrated database seeds, so you're going to have to wait a bit while we create a blank database...


Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, morango, auth, content, contenttypes, exams, logger
Running migrations:
  Rendering model states... DONE
  Applying kolibriauth.0001_initial... OK
  Applying contenttypes.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0001_initial... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying content.0001_initial... OK
  Applying content.0002_channelmetadatacache_last_updated... OK
  Applying content.0003_auto_20170607_1212... OK
  Applying exams.0001_initial... OK
  Applying kolibriauth.0002_auto_20170608_2125... OK
  Applying kolibriauth.0003_auto_20170621_0958... OK
  Applying logger.0001_initial... OK
  Applying logger.0002_auto_20170518_1031... OK
  Applying logger.0003_auto_20170531_1140... OK
  Applying morango.0001_initial... OK
  Applying morango.0002_auto_20170511_0400... OK
  Applying morango.0003_auto_20170519_0543... OK
  Applying morango.0004_auto_20170520_2112... OK
  Applying morango.0005_auto_20170629_2139... OK
  Applying morango.0006_instanceidmodel_system_id... OK
  Applying sessions.0001_initial... OK
Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, morango, auth, content, contenttypes, exams, logger
Running migrations:
  Rendering model states... DONE
  Applying kolibriauth.0001_initial... OK
  Applying contenttypes.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0001_initial... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying content.0001_initial... OK
  Applying content.0002_channelmetadatacache_last_updated... OK
  Applying content.0003_auto_20170607_1212... OK
  Applying exams.0001_initial... OK
  Applying kolibriauth.0002_auto_20170608_2125... OK
  Applying kolibriauth.0003_auto_20170621_0958... OK
  Applying logger.0001_initial... OK
  Applying logger.0002_auto_20170518_1031... OK
  Applying logger.0003_auto_20170531_1140... OK
  Applying morango.0001_initial... OK
  Applying morango.0002_auto_20170511_0400... OK
  Applying morango.0003_auto_20170519_0543... OK
  Applying morango.0004_auto_20170520_2112... OK
  Applying morango.0005_auto_20170629_2139... OK
  Applying morango.0006_instanceidmodel_system_id... OK
  Applying sessions.0001_initial... OK
INFO     Automatically enabling applications.
INFO:kolibri.utils.cli:Automatically enabling applications.
INFO     Running update routines for new version...
INFO:kolibri.utils.cli:Running update routines for new version...
Copying '/home/osboxes/.pex/install/kolibri-0.5.0b1-py2.py3-none-any.whl.b0cce2f7791692dbc00cbf6d667637d4c24cc0b5/kolibri-0.5.0b1-py2.py3-none-any.whl/kolibri/core/static/default_frontend/3-0.5.0beta1.js'
Copying '/home/osboxes/.pex/install/kolibri-0.5.0b1-py2.py3-none-any.whl.b0cce2f7791692dbc00cbf6d667637d4c24cc0b5/kolibri-0.5.0b1-py2.py3-none-any.whl/kolibri/core/static/default_frontend/4-0.5.0beta1.js'

...

Copying '/home/osboxes/.pex/install/kolibri-0.5.0b1-py2.py3-none-any.whl.b0cce2f7791692dbc00cbf6d667637d4c24cc0b5/kolibri-0.5.0b1-py2.py3-none-any.whl/kolibri/dist/kolibri_exercise_perseus_plugin/static/mathjax/2.1/config/KAthJax-e76365d62c31e1cbafe939bf94aa8497.js'

377 static files copied to '/home/osboxes/.kolibri/static'.
Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, morango, auth, content, contenttypes, exams, logger
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, morango, auth, content, contenttypes, exams, logger
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
INFO     Running 'kolibri start' as daemon (system service)
INFO:kolibri.utils.cli:Running 'kolibri start' as daemon (system service)
INFO     Going to daemon mode, logging to /home/osboxes/.kolibri/server.log
INFO:kolibri.utils.cli:Going to daemon mode, logging to /home/osboxes/.kolibri/server.log
Exception in thread MESSAGEPROCESSOR-bluebird-uniform-yellow-fifteen (most likely raised during interpreter shutdown):Exception in thread WORKERMESSAGEHANDLER-missouri-pip-football-moon (most likely raised during interpreter shutdown):Exception in thread SCHEDULER-wolfram-mobile-victor-gee (most likely raised during interpreter shutdown):

Traceback (most recent call last):Traceback (most recent call last):

  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner


  File "/home/osboxes/.pex/install/kolibri-0.5.0b1-py2.py3-none-any.whl.b0cce2f7791692dbc00cbf6d667637d4c24cc0b5/kolibri-0.5.0b1-py2.py3-none-any.whl/kolibri/dist/barbequeue/common/utils.py", line 53, in runTraceback (most recent call last):  File "/home/osboxes/.pex/install/kolibri-0.5.0b1-py2.py3-none-any.whl.b0cce2f7791692dbc00cbf6d667637d4c24cc0b5/kolibri-0.5.0b1-py2.py3-none-any.whl/kolibri/dist/barbequeue/common/utils.py", line 53, in run


  File "/usr/lib/python2.7/threading.py", line 614, in wait  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner  File "/usr/lib/python2.7/threading.py", line 614, in wait


  File "/usr/lib/python2.7/threading.py", line 355, in wait  File "/home/osboxes/.pex/install/kolibri-0.5.0b1-py2.py3-none-any.whl.b0cce2f7791692dbc00cbf6d667637d4c24cc0b5/kolibri-0.5.0b1-py2.py3-none-any.whl/kolibri/dist/barbequeue/common/utils.py", line 53, in run  File "/usr/lib/python2.7/threading.py", line 355, in wait


<type 'exceptions.TypeError'>: 'NoneType' object is not callable  File "/usr/lib/python2.7/threading.py", line 614, in wait<type 'exceptions.TypeError'>: 'NoneType' object is not callable


  File "/usr/lib/python2.7/threading.py", line 355, in wait
<type 'exceptions.TypeError'>: 'NoneType' object is not callable
osboxes@osboxes:~/Desktop$ 

```

When I restart and try to run the PEX again:

```
osboxes@osboxes:~/Desktop$ ./kolibri-v0.5.0-beta1.pex start
INFO     Version was 0.5.0beta1
, new version: 0.5.0beta1

INFO     Running update routines for new version...

0 static files copied to '/home/osboxes/.kolibri/static', 377 unmodified.
Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, morango, auth, content, contenttypes, exams, logger
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, morango, auth, content, contenttypes, exams, logger
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
INFO     Running 'kolibri start' as daemon (system service)
INFO:kolibri.utils.cli:Running 'kolibri start' as daemon (system service)
INFO     Going to daemon mode, logging to /home/osboxes/.kolibri/server.log
INFO:kolibri.utils.cli:Going to daemon mode, logging to /home/osboxes/.kolibri/server.log
WARNING:barbequeue.common.utils.MESSAGEPROCESSOR[eleven-india-pasta-colorado]:Got an exception running <bound method WorkerBackend.process_messages of <barbequeue.worker.backends.inmem.WorkerBackend object at 0x7ff6872b77d0>>: 'NoneType' object has no attribute 'Empty'
osboxes@osboxes:~/Desktop$ 

```

On Windows we have the following issue with this release: installation finishes OK, server starts without problem (Win 8.1 VM), but then it gets stuck at creating the device owner or facility:

![win8 1 start running - oracle vm virtualbox_309](https://user-images.githubusercontent.com/1457929/27887080-236fbcd6-61df-11e7-8e0d-1badcac81de4.png)

No errors in the console, command line output:

```
Copying 'C:\Python27\lib\site-packages\kolibri\dist\kolibri_exercise_perseus_plu
gin\static\mathjax\2.1\jax\output\HTML-CSS\fonts\TeX\WinIE6\Regular\Main.js'

378 static files copied to 'C:\Users\IEUser\.kolibri\static'.
Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, morango, auth, content, co
ntenttypes, exams, logger
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won
't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage
.py migrate' to apply them.
Operations to perform:
  Apply all migrations: sessions, admin, kolibriauth, morango, auth, content, co
ntenttypes, exams, logger
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won
't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage
.py migrate' to apply them.
←[37mINFO     Running 'kolibri start' in foreground...←[0m
INFO:kolibri.utils.cli:Running 'kolibri start' in foreground...
INFO:cherrypy.error:[05/Jul/2017:15:00:57] ENGINE Bus STARTING
INFO:cherrypy.error:[05/Jul/2017:15:00:57] ENGINE Started monitor thread '_Timeo
utMonitor'.
INFO:cherrypy.error:[05/Jul/2017:15:00:58] ENGINE Serving on http://0.0.0.0:8080

INFO:cherrypy.error:[05/Jul/2017:15:00:58] ENGINE Bus STARTED
INFO:cherrypy.access.35116848:127.0.0.1 - - [05/Jul/2017:15:01:41] "GET /static/
default_frontend/default_frontend-0.5.0beta1.js HTTP/1.1" 200 1218793 "http://12
7.0.0.1:8080/setup_wizard/" "Mozilla/5.0 (Windows NT 6.3) AppleWebKit/537.36 (KH
TML, like Gecko) Chrome/55.0.2883.75 Safari/537.36"
INFO:cherrypy.access.35116848:127.0.0.1 - - [05/Jul/2017:15:01:41] "GET /static/
setup_wizard/setup_wizard-0.5.0beta1.js HTTP/1.1" 200 29630 "http://127.0.0.1:80
80/setup_wizard/" "Mozilla/5.0 (Windows NT 6.3) AppleWebKit/537.36 (KHTML, like
Gecko) Chrome/55.0.2883.75 Safari/537.36"
INFO:cherrypy.access.35116848:127.0.0.1 - - [05/Jul/2017:15:01:44] "GET /static/
setup_wizard/logo-min.png?eb2802030a74ce18b835e13478e6c62e HTTP/1.1" 200 13424 "
http://127.0.0.1:8080/setup_wizard/" "Mozilla/5.0 (Windows NT 6.3) AppleWebKit/5
37.36 (KHTML, like Gecko) Chrome/55.0.2883.75 Safari/537.36"
INFO:cherrypy.access.35116848:127.0.0.1 - - [05/Jul/2017:15:01:45] "GET /static/
images/logo.ico HTTP/1.1" 200 5430 "http://127.0.0.1:8080/setup_wizard/" "Mozill
a/5.0 (Windows NT 6.3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.7
5 Safari/537.36"
Installed 2 object(s) from 1 fixture(s)
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <module>
  File "C:\Python27\lib\multiprocessing\forking.py", line 380, in main
  File "C:\Python27\lib\multiprocessing\forking.py", line 380, in main
  File "C:\Python27\lib\multiprocessing\forking.py", line 380, in main
  File "C:\Python27\lib\multiprocessing\forking.py", line 380, in main
                prepare(preparation_data)
prepare(preparation_data)
prepare(preparation_data)
prepare(preparation_data)
  File "C:\Python27\lib\multiprocessing\forking.py", line 488, in prepare
  File "C:\Python27\lib\multiprocessing\forking.py", line 488, in prepare
  File "C:\Python27\lib\multiprocessing\forking.py", line 488, in prepare
  File "C:\Python27\lib\multiprocessing\forking.py", line 488, in prepare
                assert main_name not in sys.modules, main_name
assert main_name not in sys.modules, main_name
assert main_name not in sys.modules, main_name
assert main_name not in sys.modules, main_name
AssertionErrorAssertionErrorAssertionErrorAssertionError: : : : _____main___main
___main___main__
```

Which looks just like the issue on my last test on 0.4 release (#1731).

--- a/requirements/base.txt
+++ b/requirements/base.txt
@@ -10,7 +10,7 @@ djangorestframework-csv==1.4.1
 tqdm==4.8.3                     # progress bars
 requests==2.10.0
 https://github.com/cherrypy/cherrypy/archive/v6.2.0.zip#egg=cherrypy
-https://github.com/learningequality/barbequeue/archive/e4f32.zip#egg=barbequeue
+https://github.com/learningequality/barbequeue/archive/d135b.zip#egg=barbequeue
 porter2stemmer==1.0
 unicodecsv==0.14.1
 metafone==0.5

